topostats/filters.py

```diff
@@ -431,7 +431,7 @@ def filter_image(self) -> None:
             )
         else:
             LOGGER.info(f"[{self.filename}] : Skipping scar removal as requested from config")
-            self.images["initial_scar_removal"] = self.images["initial_quadratic_removal"]
+            self.images["initial_scar_removal"] = self.images["initial_nonlinear_polynomial_removal"]
```

Comment:
In the previous `if run_scar_removal:` section we use tuple unpacking to assign the results of `scars.remove_scars()` to `self.images["initial_scar_removal"]` and `_scar_mask`.

But in this `else:` section nothing gets assigned to `_scar_mask`.

Do we need to assign an array to `_scar_mask`? If not the tuple unpacking in the `if run_scar_removal:` should perhaps be `self.images["initial_scar_removal"], _ = ...` as if `_scar_mask` isn't used I would have expected `pylint` to throw an error saying that variable was assigned but not used.

Same is true of the other change in this PR.

Reply:
Fixed 👍